### PR TITLE
feat: loading screen improvements initial load

### PR DESCRIPTION
**Pull request by moliva**
# What? 
- Refactors kernel subsystems into sagas
- Decouples lobby login from user authentication
- Allows for parallelization in initial load of client

### CODE REVIEW

File: kernel/packages/entryPoints/unity.ts [hunk 1]
```diff
@@ -47,14 +50,21 @@ initializeUnity(container)
     i.ConfigureHUDElement(HUDElementID.TERMS_OF_SERVICE, { active: true, visible: true })
     i.ConfigureHUDElement(HUDElementID.TASKBAR, { active: true, visible: true })
     i.ConfigureHUDElement(HUDElementID.WORLD_CHAT_WINDOW, { active: true, visible: true })
-    i.ConfigureHUDElement(HUDElementID.FRIENDS, { active: identity.hasConnectedWeb3, visible: false })
     i.ConfigureHUDElement(HUDElementID.OPEN_EXTERNAL_URL_PROMPT, { active: true, visible: true })
     i.ConfigureHUDElement(HUDElementID.NFT_INFO_DIALOG, { active: true, visible: false })
     i.ConfigureHUDElement(HUDElementID.TELEPORT_DIALOG, { active: true, visible: false })
     i.ConfigureHUDElement(HUDElementID.CONTROLS_HUD, { active: true, visible: false })
 
+    userAuthentified()
```
Comment (eordano) [inline kernel/packages/entryPoints/unity.ts]:
```suggestion
    try {
      await userAuthentified()
      const identity = getCurrentIdentity(globalThis.globalStore.getState())!
      i.ConfigureHUDElement(HUDElementID.FRIENDS, { active: identity.hasConnectedWeb3, visible: false })
    } catch (e) {
      logger.error('error on configuring friends hub')
    }
```

File: kernel/packages/shared/store/rootSaga.ts [hunk 2]
```diff
@@ -8,10 +8,16 @@ import { metricSaga } from './metricSaga'
...
+import { analyticsSaga } from '../analytics/sagas'
+import { friendsSaga } from '../friends/sagas'
 
 export function createRootSaga() {
   return function* rootSaga() {
     yield fork(metaSaga)
+    yield fork(analyticsSaga)
+    yield fork(friendsSaga)
+    yield fork(sessionSaga)
     yield fork(profileSaga)
```
Comment (pablitar) [inline kernel/packages/shared/store/rootSaga.ts:21]: Seems to me that this is missing one `yield fork(commsSaga)` :upside_down_face:

File: kernel/packages/shared/store/rootSaga.ts [hunk 3]
```diff
@@ -8,10 +8,16 @@ import { metricSaga } from './metricSaga'
...
+import { sessionSaga } from '../session/sagas'
+import { analyticsSaga } from '../analytics/sagas'
+import { friendsSaga } from '../friends/sagas'
 
 export function createRootSaga() {
   return function* rootSaga() {
     yield fork(metaSaga)
+    yield fork(analyticsSaga)
+    yield fork(friendsSaga)
+    yield fork(sessionSaga)
```
Comment (pablitar) [inline kernel/packages/shared/store/rootSaga.ts:21]: Seems to me that this is missing one `yield fork(commsSaga)` :upside_down_face:

File: kernel/packages/shared/comms/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  yield takeEvery(LOGIN_COMPLETED, establishCommunications)
+}
+
+function* establishCommunications() {
+  if (STATIC_WORLD) {
+    return
+  }
+
+  const identity = yield select(getCurrentIdentity)
+
```
Comment (pablitar) [inline kernel/packages/shared/comms/sagas.ts:30]: You may want to add `yield call(ensureRealmInitialized)` here too.

Comment (moliva) [reply kernel/packages/shared/comms/sagas.ts:30]: Good catch!

**Review by pbosio — APPROVED** [review]
No issues found in Builder ✅

**Review by pablitar — APPROVED** [review]
Love it!

Let's test this a lot next week :)

**Pull request merged**